From 8a8fe208169d2f78a009ddc22566b33b539aed05 Mon Sep 17 00:00:00 2001
From: xuxiang <xuxiang@eswincomputing.com>
Date: Fri, 7 Jun 2024 15:29:25 +0800
Subject: [PATCH 076/109] WIN2030-15279:fix:Adjust the pwm led cont strategy

Changelogs:
1. Adjust the pwm led control strategy
2. enable crc init

Change-Id: Ie8d06b3b761abb5f0b665b475a8f7f44939387ed
---
 Core/Src/hf_board_init.c | 2 +-
 Core/Src/hf_common.c     | 7 ++-----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/Core/Src/hf_board_init.c b/Core/Src/hf_board_init.c
index 9132ede..577efe2 100644
--- a/Core/Src/hf_board_init.c
+++ b/Core/Src/hf_board_init.c
@@ -138,7 +138,7 @@ int board_init(void)
 	MX_RTC_Init();
 	MX_TIM4_Init();
 	// MX_SPI1_Init();
-	// MX_CRC_Init();
+	MX_CRC_Init();
 	// MX_RNG_Init();
 	MX_TIM1_Init();
 	MX_TIM9_Init();
diff --git a/Core/Src/hf_common.c b/Core/Src/hf_common.c
index fd68fa2..ad52be9 100644
--- a/Core/Src/hf_common.c
+++ b/Core/Src/hf_common.c
@@ -351,7 +351,6 @@ static int get_carrier_board_info(void)
 			write_cbinfo(pgCarrier_Board_Info, cbinfo_backup);
 		}
 	}
-
 	return 0;
 }
 
@@ -1337,22 +1336,20 @@ uint32_t es_autoboot(void)
 
 void set_mcu_led_status(led_status_t type)
 {
-	printf("set_mcu_led_status type %x\n",type);
 	if( LED_MCU_RUNING == type) {
 		HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
 		HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_2);
 		HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_3);
 	}
 	else if( LED_SOM_BOOTING == type) {
-		HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_1);
 		HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_2);
 		HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_3);
 	}
 	else if( LED_SOM_KERNEL_RUNING == type)
 	{
-		if(!(HAL_TIM_ACTIVE_CHANNEL_3 & HAL_TIM_GetActiveChannel(&htim1)))
+		if(HAL_TIM_CHANNEL_STATE_BUSY == HAL_TIM_GetChannelState(&htim1, TIM_CHANNEL_2))
 		{
-			HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_1);
+			// HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_1);
 			HAL_TIM_PWM_Stop(&htim1, TIM_CHANNEL_2);
 			HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_3);
 		} else {
-- 
2.25.1

